### PR TITLE
Add sphinx directory and conf.py

### PR DESCRIPTION
**Pull request by jlin27**
Related PRs and details here: https://github.com/pytorch/serve/pull/254#issuecomment-620259092

---
This PR adds files used for the automated docs build
* Add a conf.py file in docs/
* Add a sphinx/ directory that contains build files
* Note: The Makefile uses the path .. which docs/as the source of documents

### CODE REVIEW

File: docs/sphinx/src/pip-delete-this-directory.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+This file is placed here by pip to indicate the source was put
+here by pip.
+
+Once this package is successfully installed this source code will be
```
Comment (aaronmarkham) [inline docs/sphinx/src/pip-delete-this-directory.txt:4]: Does this file need to be here? Is it related to the theme?

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+napoleon_use_ivar = True
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
+# The suffix(es) of source filenames.
+# You can specify multiple suffix as a list of string:
+#
+source_suffix = ['.rst', '.md']
```
Comment (aaronmarkham) [inline docs/conf.py:70]: I suggest changing recommonmark setup to this:
https://www.sphinx-doc.org/en/master/usage/markdown.html

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+# sys.path.insert(0, os.path.abspath('.'))
+import torch
+
+import pytorch_sphinx_theme
+
+# -- General configuration ------------------------------------------------
+
+# If your documentation needs a minimal Sphinx version, state it here.
+#
+needs_sphinx = '1.6'
```
Comment (aaronmarkham) [inline docs/conf.py:31]: Upgrade? Latest is 3.0.3...

File: docs/sphinx/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+sphinx
```
Comment (aaronmarkham) [inline docs/sphinx/requirements.txt:1]:
```suggestion
sphinx==3.0.3
```

File: docs/sphinx/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+recommonmark
```
Comment (aaronmarkham) [inline docs/sphinx/requirements.txt:2]:
```suggestion
recommonmark==0.6.0
```

File: docs/conf.py (hunk 2 above)
Comment (aaronmarkham) [inline docs/conf.py:70]:
```suggestion
source_suffix = {
    '.rst': 'restructuredtext',
    '.txt': 'markdown',
    '.md': 'markdown',
}
```

File: docs/conf.py [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+
+
+# Example configuration for intersphinx: refer to the Python standard library.
+intersphinx_mapping = {
+    'python': ('https://docs.python.org/3', None),
+    'numpy': ('https://docs.scipy.org/doc/numpy/', None),
+    'torch': ('https://pytorch.org/docs/stable/', None),
+}
+
+# Markdown parser - recommonmark
```
Comment (aaronmarkham) [inline docs/conf.py:223]: I delete this part in favor of the newer setup for recommonmark.

File: docs/conf.py [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+    'sphinx.ext.autosectionlabel',
+    'sphinx.ext.autosummary',
+    'sphinx.ext.doctest',
+    'sphinx.ext.intersphinx',
+    'sphinx.ext.todo',
+    'sphinx.ext.coverage',
+    'sphinx.ext.napoleon',
+    'sphinx.ext.viewcode',
+    'sphinxcontrib.katex',
+
```
Comment (aaronmarkham) [inline docs/conf.py:47]: Add in this so it's part of this list:
`extensions = ['recommonmark']`

File: docs/conf.py (hunk 6 above)
Comment (jlin27) [reply docs/conf.py]: Deleted

File: docs/sphinx/src/pip-delete-this-directory.txt (hunk 1 above)
Comment (jlin27) [reply docs/sphinx/src/pip-delete-this-directory.txt:4]: No this was a file generated during the build. Will remove

**Comment by maaquib** [comment]
@aaronmarkham Since this is not a documentation bug fixes (for the code which is already on master), should the base be `staging_0_1_1`

**Comment by jlin27** [comment]
@aaronmarkham - Is your team ready if I go ahead and merge this PR?

**Comment by maaquib** [comment]
@aaronmarkham @jlin27 If this is going into v0.1.1, please change the base branch to `staging_0_1_1` and I'll merge it

**Comment by jlin27** [comment]
Hi @maaquib - If master is usually your team's unstable branch and there's a plan for a new version cut, that makes sense to me. I'll change the base to `staging_0_1_1` and ping when I've updated

**Comment by jlin27** [comment]
@maaquib - Changed the base branch to `staging_0_1_1`. Please merge when the tests have passed. Thanks!

File: docs/sphinx/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,3 @@
+sphinx==3.0.3
+recommonmark==0.6.0
+-e git+git://github.com/pytorch/pytorch_sphinx_theme.git#egg=pytorch_sphinx_theme
```
Comment (aaronmarkham) [inline docs/sphinx/requirements.txt:3]: The pip install of sphinx didn't cut it... 
```
(ts_docs) /Users/markhama/Development/a2-serve/docs/sphinx markhama@f01898b363ea% make html                             
/bin/sh: sphinx-build: command not found
```
So I did a `conda install sphinx=3`
And that got me past that problem... 
I also had to install:

```
pip install sphinxcontrib-katex torch
```
And I still ran into issues with `make html`... we need at least a stub index.rst in there like...

```
############
TorchServe
############

.. toctree::
   :maxdepth: 1
   :glob:

   *
```
Even so there are a lot of warnings. 
But I think if you update the requirements and validate you can do a fresh env and get a build, then we're at a good starting point to layer on the rest.

Comment (jlin27) [reply docs/sphinx/requirements.txt:3]: Thanks for the review

* Added `sphinxcontrib-katex` and `torch`. Note they will also be added as part of the github actions
* I was able to install sphinx by running `pip install -r requirements.txt` from the docs/sphinx folder. The github actions will similarly use `pip install`.
* The index.rst file will be added in [PR #290 (https://github.com/pytorch/serve/pull/290/files)
* The warnings are syntax suggestions in the md files; they aren't breaking, but something we can clean up later if desired

Comment (jlin27) [reply docs/sphinx/requirements.txt:3]: Thanks for the review

* Added `sphinxcontrib-katex` and `torch`. Note they will also be added as part of the github actions
* I was able to install sphinx by running `pip install -r requirements.txt` from the docs/sphinx folder. The github actions will similarly use `pip install`.
* The index.rst file will be added in [PR #290 (https://github.com/pytorch/serve/pull/290/files)
* The warnings are syntax suggestions in the md files; they aren't breaking, but something we can clean up later if desired

**Comment by jlin27** [comment]
@maaquib - Reviewed and updated. Could you help merge once the tests pass? Thanks!

**Pull request merged**